<!-- Phidgets IMU launch file -->

<launch>

  #### Nodelet manager ######################################################

  <node pkg="nodelet" type="nodelet" name="imu_manager" 
    args="manager" output="screen" />

  #### IMU Driver ###########################################################

  <node pkg="nodelet" type="nodelet" name="PhidgetsImuNodelet" 
    args="load phidgets_imu/PhidgetsImuNodelet imu_manager" 
    output="screen">

    # supported data rates: 4 8 16 24 32 40 ... 1000 (in ms)
    <param name="period" value="4"/>

    # optional param serial_number, default is -1
    <!-- <param name="serial_number" value="123456"/> -->

    # compass correction params (see http://www.phidgets.com/docs/1044_User_Guide)
    <!-- <param name="cc_mag_field" value="0.52859"/>
         <param name="cc_offset0" value="0.03921"/>
         <param name="cc_offset1" value="0.19441"/>
         <param name="cc_offset2" value="-0.03493"/>
         <param name="cc_gain0" value="1.81704"/>
         <param name="cc_gain1" value="1.81028"/>
         <param name="cc_gain2" value="2.04819"/>
         <param name="cc_t0" value="0.00142"/>
         <param name="cc_t1" value="-0.03591"/>
         <param name="cc_t2" value="0.00160 "/>
         <param name="cc_t3" value="-0.05038"/>
         <param name="cc_t4" value="-0.03942"/>
         <param name="cc_t5" value="-0.05673"/> -->
  </node>

  #### IMU Orientation Filter ###############################################

  <node pkg="nodelet" type="nodelet" name="ImuFilterNodelet" 
    args="load imu_filter_madgwick/ImuFilterNodelet imu_manager" 
    output="screen">

    <param name="use_mag" value="true"/>

  </node>

</launch>
